src/dra/7E4BC.c

```diff
@@ -1217,7 +1217,261 @@ void EntityPlayerPinkEffect(Entity* self) {
+                prim->x1 = prim->x3 = prim->x0 + s3;
+            }
+            prim->y0 = prim->y1 = s7 + (s4 * (temp_i / 6));
+            prim->y2 = prim->y3 = prim->y0 + s4;
+            prim->u0 = prim->u2 = (sp4F) + (s3 * (temp_i % 6)) + 0x80;
+            prim->u1 = prim->u3 = prim->u0 + s3;
+            prim->v0 = prim->v1 = (sp4E + (s4 * (temp_i / 6)));
+            prim->v2 = prim->v3 = prim->v0 + s4;
+            prim->g0 = (((temp_i / 6) * 2) + (rand() & 3));
+            prim->tpage = 0x18;
+            prim->clut = PLAYER.palette & 0x7FFF;
+            prim->priority = PLAYER.zPriority + 2;
+            prim->drawMode = DRAW_UNK_200 | DRAW_UNK_100 | DRAW_HIDE;
+            prim = prim->next;
+        }
+        self->step += 1;
+        break;
+    case 1:
+        StoreImage(&D_800AE130, D_80139A7C);
+        prim = &g_PrimBuf[self->primIndex];
+        for (temp_i = 0; temp_i < PrimCount; temp_i++) {
+            if (self->facingLeft) {
+                prim->x0 = prim->x2 = s5 - (s3 * (temp_i % 6)) + 1;
+                prim->x1 = prim->x3 = prim->x0 - s3;
+                ;
+            } else {
+                prim->x0 = prim->x2 = s5 + (s3 * (temp_i % 6));
+                prim->x1 = prim->x3 = prim->x0 + s3;
+            }
+            prim->y0 = prim->y1 = s7 + (s4 * (temp_i / 6));
+            prim->y2 = prim->y3 = prim->y0 + s4;
+            prim = prim->next;
+        }
+        self->step += 1;
+        /* fallthrough */
+    case 2:
+        PLAYER.animCurFrame |= ~0x7FFF;
+        if (g_Player.padTapped & 0xF000) { // different between PSX and PSP
+            self->step += 1;
+        }
+        prim = &g_PrimBuf[self->primIndex];
+        for (temp_i = 0; temp_i < PrimCount; temp_i++) {
+            if (self->facingLeft) {
+                prim->x0 = prim->x2 = (s5 - (s3 * (temp_i % 6))) + 1;
+                prim->x1 = prim->x3 = prim->x0 - s3;
+            } else {
+                prim->x0 = prim->x2 = s5 + (s3 * (temp_i % 6));
+                prim->x1 = prim->x3 = prim->x0 + s3;
+            }
+            prim->y0 = prim->y1 = s7 + (s4 * (temp_i / 6));
+            prim->y2 = prim->y3 = prim->y0 + s4;
+            prim = prim->next;
+        }
+        break;
+    case 3:
+        PLAYER.animCurFrame |= ~0x7FFF;
+        temp2 = D_800AE140;
+        self->ext.dissolve.unk7E =
+            *(temp2 + self->ext.dissolve.unk80 * 8 + self->ext.dissolve.unk7C);
+        data = D_80139A7C;
+        s2 = (u8*)data;
+        s2 = s2 + ((self->ext.dissolve.unk7E >> 1) & 7);
+        s2 = s2 + (((self->ext.dissolve.unk7E & 0xFF) >> 4) << 6);
+        for (temp_i = 0; temp_i < Iterations; temp_i++) {
+            if (rand() & 3) {
+                if (self->ext.dissolve.unk7E & 1) {
+                    if (*(s2 + (temp_i & 7) * 8 + (temp_i >> 3) * 0x400)) {
+                        // odd... zeros out the location entirely
+                        *(s2 + (temp_i & 7) * 8 + (temp_i >> 3) * 0x400) &=
```

Comment:
What I think this is doing is erasing pixels in Alucard's sprite. They're 4-bit hence the 0xf0 and 0x0f. I'm not sure what color 1 in is the palette though